src/sql/src/pure.rs

```diff
@@ -152,10 +154,114 @@ pub async fn purify_statement(
         Statement::AlterSource(stmt) => {
             purify_alter_source(catalog, stmt, connection_context).await
         }
+        Statement::CreateSink(stmt) => purify_create_sink(catalog, stmt),
         o => unreachable!("{:?} does not need to be purified", o),
     }
 }
 
+pub fn purify_create_sink(
+    catalog: impl SessionCatalog,
+    mut stmt: CreateSinkStatement<Aug>,
+) -> Result<
+    (
+        Vec<(GlobalId, CreateSubsourceStatement<Aug>)>,
+        Statement<Aug>,
+    ),
+    PlanError,
+> {
+    // updating avro format with comments so that they are frozen in the `create_sql`
+    // only if the feature is enabled
+    if catalog.system_vars().enable_sink_doc_on_option() {
```

Comment:
Do we produce an error if a user uses the `DOC ON` option but the feature flag is disabled?

Reply:
yes, we do planning after this https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/plan/statement/ddl.rs#L2680-L2682

Reply:
Ah cool, thanks.